frontend/src/app/email-log-admin/email-log-admin.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { ApiService } from '../_services/api/api.service';
+import { MatSort } from '@angular/material/sort';
+import {ActivatedRoute, Router} from '@angular/router';
+import {AuthenticationService} from '../_services/api/authentication-service';
+import {EmailLog} from '../_models/email_log';
+import {User} from '../_models/user';
+import {MatTableDataSource} from '@angular/material';
+
+@Component({
+  selector: 'app-email-log-admin',
+  templateUrl: './email-log-admin.component.html',
+  styleUrls: ['./email-log-admin.component.scss']
+})
+export class EmailLogAdminComponent implements OnInit {
+  @ViewChild(MatSort, {static: true}) sort: MatSort;
+  currentUser: User;
+  dataSource: MatTableDataSource<EmailLog>;
+  displayedColumns: string[] = ['id', 'user_id', 'type', 'viewed', 'date_viewed', 'last_updated'];
+  loading = true;
+
+  constructor(
+    private api: ApiService,
+    private route: ActivatedRoute,
+    private authenticationService: AuthenticationService,
+  ) {
+    this.authenticationService.currentUser.subscribe(x => this.currentUser = x);
+  }
+
+  ngOnInit() {
+    this.api.getAllEmailLog().subscribe(log => {
```

Comment:
This is great, and will allow sorting by using the dataSource - but we really should limit it to pages, and pass the sort to the backend so we aren't trying to load, sort, and manage 10,000 emails all on the front end.